src/scripting/game_lua_kernel.cpp

```diff
@@ -3174,7 +3184,7 @@ int game_lua_kernel::intf_get_achievement(lua_State *L)
  * - Arg 2: string - id.
  * - Arg 3: int - the amount to progress the achievement.
  * - Arg 4: int - the limit the achievement can progress by
- * - Ret 1: WML table returned by the function.
+ * - Ret 1: int - the achievement's current progress after adding amount, -1 if completed, or 0 if not found
```

Comment:
Doesn't it make more sense to return max_progress if it's completed?

Reply:
That wouldn't be especially useful by itself, since the caller wouldn't know whether the value returned is the max_progress or not.

Reply:
They can obtain the max_progress separately, or you could return it as a second return value.

Reply:
So that'd be
```
 * - Ret 1: int - the achievement's current progress after adding amount or -1 if not found
 * - Ret 2: int - the achievement's max progress or 0 if not a progressable achievement
```
?